app/services/facebook_authenticator.rb

```diff
@@ -12,10 +12,14 @@ def initialize(auth_code)
   end
 
   def authenticate!
+    access_token  = get_access_token(@auth_code)
```

Comment:
Didn't really get you point to be honest but if we want it to be an instance var we should use it as such or we pass it around but then it shouldn't be an instance var.